File: Be-siukat-new/siukat-service-dev_snmptn_2022/models/log_listrik.js
var Sequelize = require('sequelize')
var db = require('../config/database')

const LogListrik = db.define('log_listrik', {
  id_log_listrik:{
  	type: Sequelize.INTEGER,
  	primaryKey: true
  },
  no_peserta: {
    type: Sequelize.STRING
  },
  no_pelanggan: {
    type: Sequelize.STRING
  },
  jenis_pemakaian: {
    type: Sequelize.ENUM('prabayar', 'pascabayar')
  },
  pengeluaran: {
    type: Sequelize.INTEGER
  },
  scan_listrik: {
    type: Sequelize.STRING
  },
  atribut: {
    type: Sequelize.ENUM('original', 'sanggah')
  },
  executor: {
    type: Sequelize.STRING
  },
  timestamp: {
    type: Sequelize.DATE
  }
},
 {
  // don't add the timestamp attributes (updatedAt, createdAt)
  timestamps: false,

  // don't delete database entries but set the newly added attribute deletedAt
  // to the current date (when deletion was done). paranoid will only work if
  // timestamps are enabled
  paranoid: true,

  // don't use camelcase for automatically added attributes but underscore style
  // so updatedAt will be updated_at
  underscored: true,

  // disable the modification of tablenames; By default, sequelize will automatically
  // transform all passed model names (first parameter of define) into plural.
  // if you don't want that, set the following
  freezeTableName: true,

  // define the table's name
  tableName: 'log_listrik'
}
);
module.exports = LogListrik

File: BackEnd-Siukat/models/log_listrik.go
package models

import "time"

type LogListrik struct {
	IDLogListrik   int        `gorm:"primaryKey;column:id_log_listrik" json:"id_log_listrik"`
	NoPeserta      string     `gorm:"column:no_peserta;type:varchar(255)" json:"no_peserta"`
	NoPelanggan    string     `gorm:"column:no_pelanggan;type:varchar(255)" json:"no_pelanggan"`
	JenisPemakaian string     `gorm:"column:jenis_pemakaian;type:enum('prabayar','pascabayar')" json:"jenis_pemakaian"`
	Pengeluaran    int        `gorm:"column:pengeluaran" json:"pengeluaran"`
	ScanListrik    string     `gorm:"column:scan_listrik;type:varchar(255)" json:"scan_listrik"`
	Atribut        string     `gorm:"column:atribut;type:enum('original','sanggah')" json:"atribut"`
	Executor       string     `gorm:"column:executor;type:varchar(255)" json:"executor"`
	Timestamp      *time.Time `gorm:"column:timestamp;type:datetime" json:"timestamp"`
}

func (LogListrik) TableName() string {
	return "log_listrik"
}
